src/components/QrcodeScannerDialog.vue

```diff
@@ -75,92 +75,123 @@
+        emit('update:modelValue', value)
       }
-    }
-  },
-  watch: {
-    cameras(cameras) {
-      if (cameras.length > 0) {
-        const cameraKey = cameras.length === 2 ? 1 : 0
-        this.currentCamera = this.cameras[cameraKey].deviceId
+    })
 
-        this.cameraStatus = 'active'
-      } else {
-        this.cameraStatus = 'nocamera'
-      }
-    },
-    currentCamera() {
-      this.scanner.start(this.currentCamera).then((content) => this.onScan(content))
-    }
-  },
-  mounted() {
-    this.init()
-  },
-  beforeUnmount() {
-    this.destroyScanner()
-  },
-  methods: {
-    init() {
-      return this.initScanner()
-        .then(() => {
-          return this.getCameras()
+    const init = async () => {
+      try {
+        scanner.value = new Scanner({
+          videoElement: videoElement.value as HTMLVideoElement
+        })
+
+        scanner.value.init().then((camerasValue: MediaDeviceInfo[]) => {
+          cameras.value = camerasValue
         })
-        .catch((err) => {
-          this.cameraStatus = 'nocamera'
-          this.$store.dispatch('snackbar/show', {
-            message: this.$t('scan.something_wrong')
-          })
-          console.error(err)
+      } catch (error) {
+        cameraStatus.value = 'nocamera'
+        store.dispatch('snackbar/show', {
+          message: t('scan.something_wrong')
         })
-    },
-    async initScanner() {
-      this.scanner = new Scanner({
-        videoElement: this.$refs.camera
-      })
-
-      return this.scanner.init()
-    },
-    destroyScanner() {
-      // First check if the scanner was initialized.
+        console.error(error)
+      }
+    }
+
+    const destroyScanner = () => {
+      // First check if the scanner controls was initialized.
       // Needed when an unexpected error occurred,
       // or when the dialog closes before initialization.
-      return this.scanner && this.scanner.stop()
-    },
-    async getCameras() {
-      this.cameras = await this.scanner.getCameras()
-    },
-    onScan(content) {
-      this.$emit('scan', content)
-      this.destroyScanner()
-      this.show = false
+      return scannerControls.value && scannerControls.value.stop()
+    }
+
+    const onScan = (content: string) => {
+      emit('scan', content)
+      destroyScanner()
+      show.value = false
+    }
+
+    watch(cameras, (cameras: MediaDeviceInfo[]) => {
+      if (cameras.length > 0) {
+        const cameraKey = cameras.length >= 2 ? 1 : 0
+        currentCamera.value = cameras[cameraKey].deviceId
+
+        cameraStatus.value = 'active'
+      } else {
+        cameraStatus.value = 'nocamera'
+      }
+    })
+
+    watch(currentCamera, async () => {
+      scannerControls.value =
+        scanner.value &&
+        (await scanner.value.start(currentCamera.value as string, (result?: Result) => {
+          if (result) {
+            onScan(result.getText()) // text is private field for zxing/browser
+          }
+        }))
+    })
```

Comment:
Simplify:

- Use `if/else` and optional chaining
- We don't need to wait until promise resolves, so you can use `void`
- Type `Result` is inferred, you can omit it.

E.g.:

```ts
if (!currentCamera.value) return;

void scanner.value?.start(currentCamera.value, result => {...})
```